File: backend/app/models/employee.rb
# app/models/employee.rb

class Employee < ApplicationRecord
  ALLOWED_SORT_COLUMNS = %w[full_name salary country job_title]
 
  validates :full_name, :job_title, :country, :salary, :currency, presence: true
  validates :salary, numericality: { greater_than: 0 }
end

File: frontend/src/types/employee.ts
export interface Employee {
  id: number;
  full_name: string;
  job_title: string;
  country: string;
  salary: number;
  currency: string;
  department: string;
}

export interface EmployeeResponse {
  data: Employee[];
  meta: {
    page: number;
    per_page: number;
    total: number;
  };
}
